Rakefile

```diff
@@ -16,10 +16,27 @@ RSpec::Core::RakeTask.new(:rails_spec) do |t|
 end
 desc 'Run rails tests'
 
-RSpec::Core::RakeTask.new(:sinatra_spec) do |t|
-  t.rspec_opts = '--require ./test/support/dummy_sinatra_api/spec/spec_helper.rb'
-  t.pattern = Dir.glob('test/support/dummy_sinatra_api/spec/*_spec.rb')
+['redis', 'memory'].each do |adapter|
+  RSpec::Core::RakeTask.new("rails_spec_with_#{adapter}_adapter") do |t|
+    ENV['STORE_ADAPTER'] = adapter
+    t.rspec_opts = '--require ./test/support/dummy_api/spec/rails_helper.rb'
+    t.pattern = Dir.glob('test/support/dummy_api/spec/**/*_spec.rb')
+  end
+  desc 'Run rails tests (with #{adapter} adapter)'
+
+  RSpec::Core::RakeTask.new("sinatra_spec_with_#{adapter}_adapter") do |t|
+    ENV['STORE_ADAPTER'] = adapter
+    t.rspec_opts = '--require ./test/support/dummy_sinatra_api/spec/spec_helper.rb'
+    t.pattern = Dir.glob('test/support/dummy_sinatra_api/spec/*_spec.rb')
+  end
+  desc "Run sinatra tests (with #{adapter} adapter)"
 end
-desc 'Run sinatra tests'
 
-task default: [:test, :rails_spec, :sinatra_spec]
+task default: [
+  :test,
+  :rails_spec,
+  :sinatra_spec_with_redis_adapter,
+  :sinatra_spec_with_memory_adapter,
+  :rails_spec_with_redis_adapter,
+  :rails_spec_with_memory_adapter
```

Comment:
What is the difference between `rails_spec`,  and `rails_spec_with_redis_adapter`/`:rails_spec_with_memory_adapter`?